tests/integration-tests/tests/storage/test_efs.py

```diff
@@ -145,39 +157,134 @@ def create_efs():
+            if subnet["AvailabilityZone"] not in availability_zones_with_mount_target:
+                # One and only one mount target should be created in each availability zone.
+                depends_on_arg = {}
+                resources = list(template.resources.keys())
+                if len(resources) >= 10:
+                    # Create at most 10 resources in parallel.
+                    depends_on_arg = {"DependsOn": [resources[-10]]}
+                template.add_resource(
+                    MountTarget(
+                        f"MountTargetResourceEfs{efs_index}Subnet{subnet_index}",
+                        FileSystemId=efs_id,
+                        SubnetId=subnet["SubnetId"],
+                        SecurityGroups=[Ref(security_group)],
+                        **depends_on_arg,
+                    )
+                )
+                availability_zones_with_mount_target.add(subnet["AvailabilityZone"])
+
+
+@pytest.fixture(scope="class")
+def mount_target_stack_factory(cfn_stacks_factory, request, region, vpc_stack):
+    """
+    EFS mount target stack.
+
+    Create mount targets in all availability zones of vpc_stack
+    """
+    created_stacks = []
+
+    def create_mount_targets(efs_ids):
+        template = Template()
+        template.set_version("2010-09-09")
+        template.set_description("Mount targets stack")
+
+        # Create security group that all communication between the mount targets and instances in the VPC
+        vpc_id = vpc_stack.cfn_outputs["VpcId"]
+        security_group = template.add_resource(
+            ec2.SecurityGroup(
+                "SecurityGroupResource",
+                GroupDescription="custom security group for EFS mount targets",
+                VpcId=vpc_id,
+            )
+        )
+        # Allow inbound connection though NFS port within the VPC
+        cidr_block_association_set = boto3.client("ec2").describe_vpcs(VpcIds=[vpc_id])["Vpcs"][0][
+            "CidrBlockAssociationSet"
+        ]
+        for index, cidr_block_association in enumerate(cidr_block_association_set):
+            vpc_cidr = cidr_block_association["CidrBlock"]
+            template.add_resource(
+                ec2.SecurityGroupIngress(
+                    f"SecurityGroupIngressResource{index}",
+                    IpProtocol="-1",
+                    FromPort=2049,
+                    ToPort=2049,
+                    CidrIp=vpc_cidr,
+                    GroupId=Ref(security_group),
+                )
+            )
+
+        # Create mount targets
+        subnet_ids = []
+        for key, value in vpc_stack.cfn_outputs.items():
+            if key.endswith("SubnetId"):
+                subnet_ids.append(value)
+        _add_mount_targets(subnet_ids, efs_ids, security_group, template)
+
+        stack = CfnStack(
+            name=generate_stack_name("integ-tests-mount-targets", request.config.getoption("stackname_suffix")),
```

Comment:
If you move this parameter outside and call it `stack_name` you can collapse this call to `CfnStack(...)` to a single line which is easier to read.

Reply:
Haha, Done!